src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/TaintedDataAnalysis.TaintedDataOperationVisitor.cs

```diff
@@ -426,22 +432,14 @@ private void TrackTaintedDataEnteringSink(SymbolAccess sink, IEnumerable<SinkKin
             /// Determines if tainted data is entering a sink as a method call or constructor argument, and if so, flags it.
             /// </summary>
             /// <param name="targetMethod">Method being invoked.</param>
-            /// <param name="taintedArguments">Arguments with tainted data to the method.</param>
+            /// <param name="visitedArguments">Arguments to the method.</param>
             /// <param name="originalOperation">Original IOperation for the method/constructor invocation.</param>
             private void ProcessTaintedDataEnteringInvocationOrCreation(
                 IMethodSymbol targetMethod,
-                IEnumerable<IArgumentOperation> taintedArguments,
+                ImmutableArray<IArgumentOperation> visitedArguments,
                 IOperation originalOperation)
             {
-                if (this.IsMethodArgumentASink(targetMethod, taintedArguments, out HashSet<SinkKind> sinkKinds))
-                {
-                    foreach (IArgumentOperation taintedArgument in taintedArguments)
-                    {
-                        TaintedDataAbstractValue abstractValue = this.GetCachedAbstractValue(taintedArgument);
-                        this.TrackTaintedDataEnteringSink(targetMethod, originalOperation.Syntax.GetLocation(), sinkKinds, abstractValue.SourceOrigins);
-                    }
-                }
-
+                EnterAMethodArgumentSink(targetMethod, visitedArguments, originalOperation);
                 if (this.TryGetInterproceduralAnalysisResult(originalOperation, out TaintedDataAnalysisResult subResult)
                     && !subResult.TaintedDataSourceSinks.IsEmpty)
```

Comment:
I think they should just work, but please add some interprocedural unit tests. E.g.

```csharp
using System.IO;
using System.Security.Cryptography;
class TestClass
{
    public byte[] GetArray(int count)
    {
        return new byte[count];
    }
    
    public void TestMethod(int offset, int count, Stream stream, CryptoStreamMode mode)
    {
        var decryptor = new AesCng().CreateDecryptor(); 
        var cryptoStream = new CryptoStream(stream, decryptor, mode);
        cryptoStream.Write(GetArray(count), offset, count);
    }

    public void DoDecryption(byte[] bytes, CryptoStream cryptoStream)
    {
        cryptoStream.Write(bytes, 0, bytes.Length);
    }
}
```